### PR TITLE
Refactor CCC UI to allow processing multiple images

### PR DESCRIPTION
**Pull request by gdetrez**
This is a first refactoring of the UI part that deals with uploading image for CCC.
It is functional and already allow the user to process multiple images but there are some things that remain to address:
- it only let you process the first plate, independently of how many plates there are in the fixture
- there might be a race condition between the plate transformation and colony detection, but only if the gridding works on the first try and if the user is very fast to click on the "Next-step" button

I'd also like to have it look less crappy if I have time :cowboy_hat_face: 

But since it's already a lot of changes, it might  be easier to review this part first :stuck_out_tongue_winking_eye:

### CODE REVIEW

File: scanomatic/ui_server_data/CCC.html [hunk 1]
```diff
@@ -60,70 +60,8 @@
...
-                                    </div>
-                                </div>
-                            </div>
-                        </div>
-                    </div>
-                    <div id="react-root"></div>
-                </div>
-            </div>
         </div>
+        <div id="react-root"></div>
```
Comment (local-minimum) [inline scanomatic/ui_server_data/CCC.html]: Not much left to remove now :)

File: scanomatic/ui_server_data/js/ccc/components/CCCEditor.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        );
+    } else {
+        const image = props.images[props.currentImage];
+        view = (
+            <PlateEditorContainer
+                pinFormat={props.pinFormat}
+                accessToken={props.accessToken}
+                cccId={props.cccId}
+                imageId={image.id}
+                plateId="1"
```
Comment (local-minimum) [inline scanomatic/ui_server_data/js/ccc/components/CCCEditor.js]: So if we need ability to have more than one plate per image, we need to deal with this hard-coded value?

File: scanomatic/ui_server_data/js/ccc/components/CCCEditor.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                onFinish={props.onFinishImage}
+            />
+        );
+    }
+
+    return (
+        <div>
+            <PlateList plates={props.images} />
```
Comment (local-minimum) [inline scanomatic/ui_server_data/js/ccc/components/CCCEditor.js]: This sounds more like an `ImageList` than a `PlateList`?

File: scanomatic/ui_server_data/js/ccc/components/ImageUpload.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+export default class ImageUpload extends React.Component {
+    constructor(props) {
+        super(props);
+        this.handleFileChange = this.handleFileChange.bind(this);
+    }
+
+    handleFileChange(event) {
+        const { target: { files } } = event;
+        this.props.onImageChange(files[0]);
+    }
```
Comment (local-minimum) [inline scanomatic/ui_server_data/js/ccc/components/ImageUpload.js]: kind of breaks the container - component pattern doesn't it? but I see the point of having it here.

File: scanomatic/ui_server_data/js/ccc/components/PlateList.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export default function PlateList(props) {
+    return (
+        <div>
+          <h4>Uploaded Images</h4>
+          <ul>
```
Comment (local-minimum) [inline scanomatic/ui_server_data/js/ccc/components/PlateList.js]: maybe just make it a list-group?
https://getbootstrap.com/docs/3.3/components/#list-group

File: scanomatic/ui_server_data/js/ccc/helpers.js [hunk 6]
```diff
@@ -167,3 +170,26 @@ export function loadImage(url) {
...
+        ];
+        return API.SetCccImageData(ccc, imageId, token, imageData, fixture);
+    }).then(() => {
+        return API.SetCccImageSlice(ccc, imageId, token);
+    }).then(() => {
+        return API.SetGrayScaleImageAnalysis(ccc, imageId, token);
+    }).then(() => {
+        return imageId;
+    });
+}
```
Comment (local-minimum) [inline scanomatic/ui_server_data/js/ccc/helpers.js]: :)

File: scanomatic/ui_server_data/js/ccc/index.js [hunk 7]
```diff
@@ -3,80 +3,15 @@ import ReactDOM from 'react-dom';
...
 } from './api';
-import { createScope, getCurrentScope, setCurrentScope } from './scope';
-import PlateEditorContainer from './containers/PlateEditorContainer';
+
+import CCCEditorContainer from './containers/CCCEditorContainer';
 
 
 
 window.cccFunctions = {
-    setStep: (step) => {
```
Comment (local-minimum) [inline scanomatic/ui_server_data/js/ccc/index.js]: Yay it is gone!

File: scanomatic/ui_server_data/js/ccc/components/CCCEditor.js (hunk 2 above)
Comment (gdetrez) [reply scanomatic/ui_server_data/js/ccc/components/CCCEditor.js]: Yes.

File: scanomatic/ui_server_data/js/ccc/components/ImageUpload.js (hunk 4 above)
Comment (gdetrez) [reply scanomatic/ui_server_data/js/ccc/components/ImageUpload.js]: I don't think so. Container are about state and side effects, this just massages data for the callback.

**Comment by gdetrez** [comment]
I'm working on refactoring `PlateList` and supporting multiple plates, so I'm not sure it's worth fixing here...

**Pull request merged**